### PR TITLE
pkg/trace/agent: use shared agent logger

### PR DESCRIPTION
**Pull request by gbbr**
This change removes the custom logger implementation and switches to
using the shared library logger. As a result, support for
`apm_config.log_throttling` is removed and instead we gain support for
full logging features such as `log_to_console` and `log_to_syslog`.

Log throttling is not something that users need to manually configure
and should be taken care of by developers to ensure that errors and
warnings never spam the user. A new package `pkg/trace/logutil` has
been added which exposes a throttled logger that allows to limit how
many times a certain message appears during a time window.

### CODE REVIEW

File: pkg/trace/logutil/once.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+// Warn logs the message at the warning level.
+func (tl *ThrottledLogger) Warn(format string, params ...interface{}) {
+	tl.log(log.Warnf, format, params...)
+}
+
+// Write implements io.Writer.
+func (tl *ThrottledLogger) Write(p []byte) (n int, err error) {
+	tl.Error(string(p))
+	return len(p), nil
```
Comment (remeh) [inline pkg/trace/logutil/once.go]: That's a detail but returning `len(p)` could be wrong here. Not sure we really want to do something about that though.

**Review by remeh — COMMENTED** [review]
That looks great!
Some notes: 
 - `once.go` is probably due to organic dev but is now a weird name for the throtted logger filename :)
 - that could be useful for other parts of the Agents, maybe we should share it in `pkg/util/log`? If doable
Also left a small comment on the `Write` method of ThrottledLogger.

File: pkg/trace/logutil/once.go (hunk 1 above)
Comment (gbbr) [reply pkg/trace/logutil/once.go]: I know, but I don't think there's much we can do about it. Open to ideas...

**Comment by gbbr** [comment]
I'm happy to move it to that package if you think it could be useful...

**Comment by gbbr** [comment]
@remeh 

* Renamed the file to `throttled.go`
* About making it shared, I'd prefer to leave that for another PR because of two reasons: it doesn't seem needed today and it wasn't until now _and_ if it's going to be shared, I'd have to polish it a lot more and add more logging levels.

Happy to make it shared but I think that needs a bit more work than this. For now it just satisfies trace-agent needs. Would that be ok? Happy to commit to it...

**Comment by remeh** [comment]
@gbbr There is no strong needs to "share" it now, we could take care of that later if the need appears. 👍

File: pkg/trace/writer/stats.go [hunk 2]
```diff
@@ -41,6 +42,8 @@ type StatsWriter struct {
 	senders  []*sender
 	stop     chan struct{}
 	stats    *info.StatsWriterInfo
+
+	easylog *logutil.ThrottledLogger
```
Comment (knusbaum) [inline pkg/trace/writer/stats.go]: Is there a reason these are named `easylog`? I don't have a real issue with it, I just find it a bit confusing. The `easy` seems like it's trying to imply something, but I don't see it explained anywhere.

**Review by knusbaum — APPROVED** [review]
LGTM. Just a couple questions/nits.

**Comment by gbbr** [comment]
@knusbaum I've merged because the comments weren't blockers and code freeze is coming, but I will address any follow up comments in a new PR as soon as I hear back from you, if you still think they are needed.

**Pull request merged**